modules/gateway/upnp.go

```diff
@@ -61,7 +66,7 @@ func (g *Gateway) learnHostname(port string) {
 		return
 	}
 
-	addr := modules.NetAddress(net.JoinHostPort(host, port))
+	addr := modules.NetAddress(net.JoinHostPort(host, g.port))
```

Comment:
Still needs an RLock, for convention immutable fields either need to be named as such (`g.immutablePort`) or they need to be under lock when being accessed/modified.

Otherwise the developer needs to remember whether or not the field is immutable, and that greatly increases the barrier to entry for navigating something as complicated as the gateway.